endpoints/openrtb2/auction.go

```diff
@@ -1391,22 +1394,47 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 					err = fmt.Errorf("imp.ext.prebid.storedrequest.id %s: Stored Imp has Invalid JSON: %s", impIds[i], Err)
 				}
 			}
-			return nil, []error{err}
+			return nil, impToStoredReq, []error{err}
 		}
 		imps[idIndices[i]] = resolvedImp
+
+		includeStoredImps, err := jsonparser.GetBoolean(resolvedImp, "ext", "prebid", "options", "echovideoattrs")
+
+		if err != nil && err != jsonparser.KeyPathNotFoundError {
+			return nil, nil, []error{err}
+		}
+
+		//determine if this is video impression
+		_, _, offset, err := jsonparser.Get(resolvedImp, "video")
+		if err != nil && err != jsonparser.KeyPathNotFoundError {
+			return nil, nil, []error{err}
+		}
+
+		if includeStoredImps && offset != -1 {
+			impId, err := jsonparser.GetString(resolvedImp, "id")
```

Comment:
A Note On `jsonparser`:
This is substantially faster for reading values from a json blob than the Go json package, but you need to keep in mind that each `jsonparser.GetXXX` call re-parses the entire json. Based on performance measurements, the tipping point of efficiency is around 4 calls.  At that point, please consider switching to `EachKey` to use a single pass.

Reply:
Good point. We are about to reach 4 calls. 
The very last call takes `storedImps[impIds[i]]` as input, 3 others `resolvedImp`
I added a comment about this.